### PR TITLE
Relay mutation

### PR DESCRIPTION
**Pull request by lyw0149**
Add graphql-relay.

### CODE REVIEW

File: src/server/app.js [hunk 1]
```diff
@@ -1,22 +1,26 @@
...
+app.use(bodyParser.json());
 
 app.post('/graphql', jwtUtil.apiProtector, graphqlHTTP((request) => {
   const startTime = Date.now();
   return {
     schema: schema,
     graphiql: true,
     rootValue: { request },
-    extensions() {
+    extensions(ext) {
```
Comment (ace0625) [inline src/server/app.js:23]: Can you explain what the ext stands for

File: src/server/util/jwt.util.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ const JWT_SECRET_KEY = process.env.JWT_SECRET_KEY;
 
 export default {
   apiProtector(req, res, next) {
+    console.log(req.body);
```
Comment (ace0625) [inline src/server/util/jwt.util.js:9]: Delete this.

File: src/shared/schema.json [hunk 3]
```diff
@@ -172,46 +172,18 @@
           "description": "Root Mutation of the Recbook Schema",
           "fields": [
             {
-              "name": "createUser",
-              "description": "Create a new user and get accessToken.",
+              "name": "_createUser",
```
Comment (ace0625) [inline src/shared/schema.json:175]: Just curious. Are you going to keep using user score?

Comment (lyw0149) [reply src/shared/schema.json:175]: I will delete schema.json from this repo.

File: src/server/util/jwt.util.js (hunk 2 above)
Comment (lyw0149) [reply src/server/util/jwt.util.js:9]: Thanks!

File: src/server/app.js (hunk 1 above)
Comment (lyw0149) [reply src/server/app.js:23]: it is in graphql-express check this out.
[express-graphql](https://github.com/graphql/express-graphql)

File: src/server/graphQLschema/mutation/user.mutation.js [hunk 4]
```diff
@@ -19,58 +22,70 @@ import User from '../type/user.type';
...
-    resolve: (source, { email, password }, _) => (
-      new Promise((resolve, reject) => {
+    outputFields: {
+      user: {
+        type: User,
+        resolve: (payload) => payload,
+      },
+    },
+    mutateAndGetPayload: ({ email, password }) => {
+      console.log(email, password);
```
Comment (festinalente91) [inline src/server/graphQLschema/mutation/user.mutation.js:65]: remove console.log.

File: src/shared/schema.graphqls [hunk 5]
```diff
@@ -3,22 +3,29 @@ schema {
...
-  getToken(email: String!, password: String!): User
+  _createUser(input: _createUserInput!): _createUserPayload
 }
 
 # Root Query of the Recbook Schema
 type rootQuery {
   # List of All users in Recbook
   users: [User]
-
   # UserType by _id
```
Comment (festinalente91) [inline src/shared/schema.graphqls:29]: Why don't we just use id rather than _id?

**Comment by ochanje210** [comment]
why did you close this

**Pull request closed without merging**